lib/fs/util.go

```diff
@@ -89,3 +90,63 @@ func IsParent(path, parent string) bool {
 	}
 	return strings.HasPrefix(path, parent)
 }
+
+// https://www.rosettacode.org/wiki/Find_common_directory_path#Go
+func CommonPrefix(paths ...string) string {
+	// Handle special cases.
+	switch len(paths) {
+	case 0:
+		return ""
+	case 1:
+		return path.Clean(paths[0])
+	}
+
+	// Note, we treat string as []byte, not []rune as is often
+	// done in Go. (And sep as byte, not rune). This is because
+	// most/all supported OS' treat paths as string of non-zero
+	// bytes. A filename may be displayed as a sequence of Unicode
+	// runes (typically encoded as UTF-8) but paths are
+	// not required to be valid UTF-8 or in any normalized form
+	// (e.g. "é" (U+00C9) and "é" (U+0065,U+0301) are different
+	// file names.
+	c := []byte(path.Clean(paths[0]))
```

Comment:
Use `filepath.Clean` (also below) to use `os.PathSeparator` not always `/` (and fiilepath is already imported).

Reply:
Sorry, this comment is a bit unclear. I guess you are saying just to use filepath rather than path?

Reply:
Yes. The package `filepath` (well `path/filepath`) is a system-specific variant of `path`, i.e. `path` always expects slash as path separator, while `filepath` uses `os.PathSeparator`.

Reply:
Man, this common path problem is much harder than I expected. relative vs abs paths, "" vs "/" vs "." etc. Edge case galore.